src/main/java/Jarvis/Deadline.java

```diff
@@ -0,0 +1,74 @@
+package Jarvis;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * Represents a deadline task that extends the Task class.
+ * <p>
+ *     The Deadline class is a subclass of the Task class and has 2 additional attributes, by and dateby.
+ *     By represents the time to complete the task by.
+ *     dateBy is by in a Java LocalDate if the formatting of by is correct.
+ * </p>
+ */
+public class Deadline extends Task {
+    protected String by;
```

Comment:
These names seem to have the potential to confuse, perhaps a more descriptive name will make the differences between the two variables clearer?